guiEditor/src/guiEditor.ts

```diff
@@ -61,15 +61,16 @@ export class GUIEditor {
         ReactDOM.render(graphEditor, hostElement);
         // create the middle workbench canvas
         if (!globalState.guiTexture) {
-            globalState.workbench.createGUICanvas();
-            if (options.currentSnippetToken) {
-                try {
-                    globalState.workbench.loadFromSnippet(options.currentSnippetToken);
+            globalState.workbench.createGUICanvas().then(() => {

```

Comment:
try using await and mark Show(...) as async so that the code might be easier to understand and ppl relying on guiEditor would be sure when everything is ok

Reply:
This was my suggestion - not to use await in this function and not changing its signature while still allowing async operation to be logged correctly.

Reply:
I wonder if we should do it while we do not have back compat issue to deal with so far ? mainly in case we have more async part during init in the future ?

Reply:
For clarification, Should I make the change? I do like the ability to be able to log should something go wrong.

Reply:
I let @RaananW chose for this one :-) not knowing enough of the rest of the project

Reply:
@sebavan is right - there is no issue with adding async to the show function.
Make the change - the code will be cleaner

Reply:
How's this. :)